gpt_json/gpt.py

```diff
@@ -117,15 +120,7 @@ async def run(
             for message in messages
         ]
 
-        # Most requests succeed on the first try but we wrap it locally here in case
-        # there is some temporarily instability with the API. If there are longer periods
-        # of instability, there should be system-wide retries in a daemon.
-        backoff_request_submission = backoff.on_exception(
-            backoff.expo,
-            (RateLimitError, OpenAITimeout, APIConnectionError),
-            max_tries=self.openai_max_retries,
-            on_backoff=handle_backoff
-        )(self.submit_request)
+        backoff_request_submission = await self._backoff_submit_request()
```

Comment:
The triple indirection of `_backoff_submit_request -> backoff -> self.submit_request makes` this path hard to follow. Is this refactor required for some codepath downstream?